tests/abi/tools/revng-check-compatibility-with-abi/Verify.cpp

```diff
@@ -446,21 +446,16 @@ getPrototypeLayout(const model::Function &Function, model::ABI::Values ABI) {
   const model::Type *Prototype = Function.Prototype().getConst();
   revng_assert(Prototype != nullptr);
   if (auto *CABI = llvm::dyn_cast<model::CABIFunctionType>(Prototype)) {
-    // Copy the prototype since we might have to modify it before testing.
-    model::CABIFunctionType PrototypeCopy = *CABI;
-
-    if (ABI != PrototypeCopy.ABI()) {
-      // Workaround for dwarf sometimes misdetecting a very specific ABI,
-      // for example, it labels `SystemV_x86_regparam_N` as just `SystemV_x86`
-      // despite them being incompatible.
-      std::array AllowedABIs = { model::ABI::SystemV_x86_regparm_3,
-                                 model::ABI::SystemV_x86_regparm_2,
-                                 model::ABI::SystemV_x86_regparm_1 };
-      revng_assert(llvm::is_contained(AllowedABIs, ABI));
-      PrototypeCopy.ABI() = ABI;
```

Comment:
Yeah, this is not good news, just my bad. Last time when we discussed this and came up with a [cleaner solution](https://github.com/revng/revng/blob/develop/share/revng/test/tests/abi/replace-abi.py), I somehow missed this and it also got merged. Practically, this was a no-op all along (since all the cases it is designed to handle were already handled by the python script).

Reply:
Uhm OK.